astropy/stats/bayesian_blocks.py

```diff
@@ -340,6 +340,8 @@ def fit(self, t, x=None, sigma=None):
         # Compute ncp_prior if not defined
         if self.ncp_prior is None:
             ncp_prior = self.compute_ncp_prior(N)
+        else:
```

Comment:
Looking at the [code of `compute_ncp_prior`](https://github.com/astropy/astropy/pull/8339/files#diff-43305fbb4f5e0bdf0201d97a3f89fa3cR287) I would rather suggest to remove the if/else statement here.

Reply:
@bsipocz I added a commit to keep this` if/else` but removed the unnecessary `if` in `compute_ncp_prior` because it wasn't really computing `ncp_prior` in the case where it was already set.